File: scoutingapp/src/main/java/org/usfirst/frc/team25/scouting/data/models/PostMatch.java
package org.usfirst.frc.team25.scouting.data.models;

import android.widget.CheckBox;

import java.util.ArrayList;
import java.util.HashMap;

/**
 * Qualitative reflection on the robot's performance after a match
 * Not to be used for end game actions
 */
public class PostMatch {


    private int allianceScore;
    private boolean winRanking;
    private boolean melody;
    private boolean ensemble;
    private int defensive;
    private int speed;
    private int manuver;
    private int hpEfficiency;
    private int ranking;
    private boolean lostComms;

    public PostMatch(int allianceScore, boolean winRanking, boolean melody, boolean ensemble, boolean lostComms, int defensive, int speed, int manuver, int hpEfficiency, int ranking) {
        this.allianceScore = allianceScore;
        this.winRanking = winRanking;
        this.melody = melody;
        this.ensemble = ensemble;
        this.lostComms = lostComms;
        this.defensive = defensive;
        this.speed = speed;
        this.manuver = manuver;
        this.hpEfficiency = hpEfficiency;
        this.ranking = ranking;
    }

    public int getAllianceScore() {
        return allianceScore;
    }
    public boolean isWinRanking() {return winRanking;}
    public boolean isMelody() {
        return melody;
    }
    public boolean isEnsemble() {
        return ensemble;
    }
    public boolean isLostComms() {return lostComms;}
    public int getDefensive() {
        return defensive;
    }
    public int getSpeed() {
        return speed;
    }
    public int getManuver() {
        return manuver;
    }
    public int getHpEfficiency() {
        return hpEfficiency;
    }
    public int getRanking() {return ranking; }

}

File: scoutingapp/src/main/java/org/usfirst/frc/team25/scouting/data/models/PostMatch.kt
package org.usfirst.frc.team25.scouting.data.models

import android.widget.CheckBox
import java.util.*

/**
 * Qualitative reflection on the robot's performance after a match
 * Not to be used for end game actions
 */
data class PostMatch(var robotComment: String, @field:Transient var robotQuickComments: ArrayList<CheckBox>,
                @field:Transient val robotQuickCommentValues: Array<String>, var focus: String, val teamOneCompare: Int,
                val teamTwoCompare: Int, val comparison: String, val pickNumber: Int) {
	private val robotQuickCommentSelections: HashMap<String, Boolean> = HashMap()
	
	fun finalizeComment() {
		for (value in robotQuickCommentValues) {
			for (cb in robotQuickComments) {
				if (cb.text.toString() == value) {
					robotQuickCommentSelections[value] = cb.isChecked
				}
			}
		}
		
		robotComment = robotComment.replace("[,\n]", ";") //prevent csv
// problems
	}
	
}